tests/test_activate.py

```diff
@@ -2315,27 +2298,31 @@ class InteractiveShell:
             # .. and this one from the `macOS` terminal:
             # pexpect.exceptions.EOF: End Of File (EOF).
             # 'args': ('-x',),
-            "init_command": 'set _CONDA_EXE="{CPR}/shell/bin/conda"; '
-            "source {CPR}/shell/etc/profile.d/conda.csh; ".format(
-                CPR=CONDA_PACKAGE_ROOT
+            "init_command": (
+                f'set _CONDA_EXE="{CONDA_PACKAGE_ROOT}/shell/bin/conda"; '
+                f"source {CONDA_PACKAGE_ROOT}/shell/etc/profile.d/conda.csh;"
             ),
             "print_env_var": 'echo "$%s"',
         },
-        "tcsh": {
-            "base_shell": "csh",
-        },
+        "tcsh": {"base_shell": "csh"},
         "fish": {
             "activator": "fish",
-            "init_command": f"eval ({exe_quoted} -m conda shell.fish hook {dev_arg})",
+            "init_command": f"eval ({EXE} -m conda shell.fish hook {dev_arg})",
             "print_env_var": "echo $%s",
         },
         # We don't know if the PowerShell executable is called
         # powershell, pwsh, or pwsh-preview.
         "powershell": {
             "activator": "powershell",
             "args": ("-NoProfile", "-NoLogo"),
-            "init_command": "{} -m conda shell.powershell hook --dev | Out-String | Invoke-Expression".format(
-                sys.executable
+            "init_command": (
+                f"{sys.executable} -m conda shell.powershell hook --dev "
+                "| Out-String "
+                "| Invoke-Expression "
+                "; conda deactivate "
+                "; conda deactivate "
+                "; conda deactivate "
+                "; conda deactivate"
```

Comment:
it's because we don't know how many envs the user/CI has activated before the tests are triggered (in other words we're trying to say `conda deactivate --all`) so this is just a random guess at a "safe" number of deactivates

the code did this previously but later on after this context was returned to the test scope, I just moved this into the startup

yes, I'll add a comment with this background 👍🏻